lib/Myriad/RPC.pm

```diff
@@ -0,0 +1,84 @@
+package Myriad::RPC;
+
+use strict;
+use warnings;
+
+# VERSION
+
+use Future::AsyncAwait;
+use Myriad::RPC::Message;
+use Object::Pad;
+
+class Myriad::RPC extends Myriad::Notifier;
+
+use experimental qw(signatures);
+
+use utf8;
+
+use Syntax::Keyword::Try;
+
+=encoding utf8;
+
+=head1 SYNOPSIS
+
+=head1 DESCRIPTION
+
+Myriad RPC implementation to serve the requests of the service clients.
+
+=cut
+
+has $redis;
+has $service;
+
+has $stream_name;
+has $group_name;
+
+has $ryu;
+has $rpc_map;
+
+method ryu {$ryu}
+method rpc_map :lvalue {$rpc_map}
+
+method configure(%args) {
+    $redis = delete $args{redis} // die 'Redis Transport is required';
```

Comment:
`->configure` can be called at various times, that's why there's a pattern of `for(KEYS) { $var = delete $args{$var} if exists $args{$var}`